aksenov/VMindbooke.Bot/Domain/ValidCollectionExtension.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
+
+namespace VMindbooke.Bot.Domain
+{
+    public static class ValidCollectionExtension
+    {
+        public static IEnumerable<T> GetValidCollection<T>(this IEnumerable<IValidObject> collection)
```

Comment:
```suggestion
        public static IEnumerable<T> GetValidCollection<T>(this IEnumerable<T> collection) where T : IValidObject
```
так тебе не придётся при использовании этого метода указывать явно тип, он будет выводиться из типа коллекции